src/main/java/org/datadog/jenkins/plugins/datadog/clients/DatadogHttpClient.java

```diff
@@ -87,35 +92,30 @@ of this software and associated documentation files (the "Software"), to deal
      */
     @SuppressFBWarnings(value="DC_DOUBLECHECK")
     public static DatadogClient getInstance(String url, String logIntakeUrl, Secret apiKey){
-        if(enableValidations){
-            if (url == null || url.isEmpty()) {
-                logger.severe("Datadog Target URL is not set properly");
-                return null;
-            }
-            if (apiKey == null || Secret.toString(apiKey).isEmpty()){
-                logger.severe("Datadog API Key is not set properly");
+        // If the configuration has not changed, return the current instance without validation
+        // since we've already validated and/or errored about the data
+
+        DatadogHttpClient newInstance = new DatadogHttpClient(url, logIntakeUrl, apiKey);
```

Comment:
Do we need to instantiate a new instance on every call?  Can we maybe embed it within the next `httpInstance != null` check, and then do a comparison inside there?

Reply:
Sure, I can try that!

Reply:
Actually, moving it won't save much except for the first time when instance is null.

Reply:
I forgot that `newInstance` is used later in the function too so we can't do this- It is actually needed more if `httpInstance` is null because we want to run validation on it https://github.com/jenkinsci/datadog-plugin/pull/59/files#diff-daeae67ef2298d24148f5336f3887d7eR110